### DISCUSSION TITLE
[Bug]: cannot import sub module of anomalib: no module named "anomalib.XXX"

### DISCUSSION THREAD

**Discussion by leopardyao** [open-edge-platform/anomalib#1699] [Q&A]
### Describe the bug

I  installed anomalib with pip install anomalib[full] with no errors.  When I type "import anomalib" I get no errors.
However, if I begin to import sub modules or run the "gettstarted" notebook,
I will get ModuleNotFoundError: No module named 'anomalib.engine' or etc.
not sure where I did wrong.

### Dataset

N/A

### Model

N/A

### Steps to reproduce the behavior

1. Create virtual environment
python -m venv  anomalib_env
2. activate the virtual enviroment
3. pip install anmalib[full]
4. I type "python"
5. in python terminal:
"from anomalib.engine import Engine"

### OS information

OS information:
- OS: Win10
- Python version: 3.8.2
- Anomalib version: 0.7.0
- PyTorch version:2.1.2
- CUDA/cuDNN version: [e.g. 11.1]
- GPU models and configuration: [e.g. 2x GeForce RTX 3090]
- Any other relevant information: [e.g. I'm using a custom dataset]

### Expected behavior

I expect I can import submodule

### Screenshots

![image](https://github.com/openvinotoolkit/anomalib/assets/35131627/dbde4100-a640-44c7-8d94-4e87116f8bcd)

### Pip/GitHub

pip

### What version/branch did you use?

main

### Configuration YAML

```yaml
I did not change any configuration
```

### Logs

```shell
No logs yet
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by blaz-r** [comment]
Hi, this is an issue as the main branch is already on v1 while the one from pip is still 0.7. Follow the instructions here https://github.com/openvinotoolkit/anomalib#-installation under tab "install from source" to get the latest v1 update where engine is available.

**Reply by leopardyao** [reply]
thanks it solves my prolem

**Comment by samet-akcay** [comment]
@leopardyao, how do you get the notebook? Download it from the main branch?

as @blaz-r pointed out, you have two options:

## Use v0.7.0

In this case, you could download anomalib via pypi. However, to use the notebooks, you need to use the specific branch. For example;
https://github.com/openvinotoolkit/anomalib/blob/v0.7.0/notebooks/000_getting_started/001_getting_started.ipynb

## Use v1.0.0dev
If you would like to try the new v1 stuff, then you will need to install the library from source 
https://github.com/openvinotoolkit/anomalib#-installation

Let us know if this fixes your issue.

**Comment by shanmugamani1023** [comment]
I tried both 0.7.0 and v1.0.0 anamalib version ,if i use 0.7.0 I will get ModuleNotFoundError: No module named 'anomalib.engine' if i use v1.0.0  i got this error ImportError: cannot import name 'AutoModelForMaskedLM' from 'transformers'
![v1 0](https://github.com/openvinotoolkit/anomalib/assets/119931738/9add929d-6f93-47dc-8397-b6cd0c024ed5)

**Comment by samet-akcay** [comment]
@shanmugamani1023, can you delete anomalib and install again? Seems like there is conflict in your versions. `Engine` stuff is new and is only available in v1. If you have anomalib v0.7.0 installed on your system, `Engine` stuff should not work.

**Comment by samet-akcay** [comment]
I'm moving this to discussion, we can continue there.
